x-pack/plugins/discover_log_explorer/public/components/dataset_selector/state_machine/state_machine.ts

```diff
@@ -118,8 +143,11 @@ export const createPureDatasetsSelectorStateMachine = (
           }
           return {};
         }),
-        storeSelected: assign((_context, event) =>
-          'dataset' in event ? { selected: event.dataset } : {}
+        storeAllSelection: assign((_context) => ({
+          selection: AllDatasetSelection.create(),
+        })),
+        storeSingleSelection: assign((_context, event) =>
+          'dataset' in event ? { selection: SingleDatasetSelection.create(event.dataset) } : {}
```

Comment:
We want to store the correct DatasetSelection instance depending on the selection mode the user can select.
These actions are triggered uniquely when a new selection is applied and work as a setter for the dataset selection strategy.